front/app/containers/Admin/routes.tsx

```diff
@@ -11,78 +13,96 @@ import createAdminMessagingRoutes from './messaging/routes';
-        } else {
-          replace(`${urlLocale && `/${urlLocale}`}/sign-in`);
-        }
-      }
-    }
+import { isNilOrError, isUUID } from 'utils/helperUtils';
+import { removeLocale } from 'utils/cl-router/updateLocationDescriptor';
+
+// typings
+import { IAppConfiguration } from 'services/appConfiguration';
+
+const isTemplatePreviewPage = (urlSegments: string[]) =>
+  urlSegments.length === 4 &&
+  urlSegments[0] === 'admin' &&
+  urlSegments[1] === 'projects' &&
+  urlSegments[2] === 'templates' &&
+  isUUID(urlSegments[3]);
+
+const getRedirectURL = (
+  appConfiguration: IAppConfiguration,
+  authUser: TAuthUser,
+  pathname: string | undefined,
+  urlLocale: string | null
+) => {
+  const localeSegment = urlLocale ? `/${urlLocale}` : '';
+
+  if (
+    appConfiguration.data.attributes.settings.core.lifecycle_stage === 'churned'
+  ) {
+    return `${localeSegment}/subscription-ended`;
+  }
+
+  if (!isNilOrError(authUser) && isModerator({ data: authUser })) {
+    return `${localeSegment}/`;
+  }
+
+  // get array with url segments (e.g. 'admin/projects/all' becomes ['admin', 'projects', 'all'])
+  const urlSegments = pathname
+    ? pathname.replace(/^\/+/g, '').split('/')
+    : null;
+
+  // check if the unauthorized user is trying to access a template preview page (url pattern: /admin/projects/templates/[id])
+  // if so, redirect them to the citizen-facing version of the template preview page (url pattern: /templates/[id])
+  // if not, redirect them to the sign-in page
+  if (urlSegments && isTemplatePreviewPage(urlSegments)) {
+    return `${localeSegment}/templates/${urlSegments[3]}`;
+  }
+
+  return `${localeSegment}/sign-in`;
+};
+
+const IndexElement = () => {
+  const location = useLocation();
+  const { pathname, urlLocale } = removeLocale(location.pathname);
+
+  const accessAuthorized = usePermission({
+    item: { type: 'route', path: pathname },
+    action: 'access',
   });
+  const appConfiguration = useAppConfiguration();
+  const authUser = useAuthUser();
+
+  if (isNilOrError(appConfiguration)) return null;
+
+  const redirectURL = accessAuthorized
+    ? null
+    : getRedirectURL(appConfiguration, authUser, pathname, urlLocale);
+
+  if (redirectURL) return <Navigate to={redirectURL} />;
+
+  return (
+    <LoadingComponent>
+      <AdminContainer />
+    </LoadingComponent>
+  );
 };
 
-const createAdminRoutes = (_isUserAuthorized: boolean) => {
+const createAdminRoutes = () => {
   return {
     path: 'admin',
-    element: (
-      <LoadingComponent>
-        <AdminContainer />
-      </LoadingComponent>
-    ),
-    // https://stackoverflow.com/questions/62384395/protected-route-with-react-router-v6
-    // onEnter: isUserAuthorized,
-    // indexRoute: {
-    //   onEnter: (nextState, replace) => {
-    //     const pathNameWithLocale = nextState.location.pathname;
-    //     const { urlLocale } = removeLocale(pathNameWithLocale);
-    //     replace(`${urlLocale && `/${urlLocale}`}/admin/dashboard`);
```

Comment:
Not very deep into this, but is this not needed anymore?

Reply:
This line: `replace(`${urlLocale && `/${urlLocale}`}/admin/dashboard`);`

Reply:
This bit now lives in `getRedirectURL`, and it's stored in the `localeSegment` variable (so that we only do it once instead of multiple times as before)

Reply:
In addition, the behavior before was actually buggy: ``${undefined && 'string'}/admin/dashboard`` evaluates to `
'undefined/admin/dashboard'`, this way it actually leaves out the locale if it doesn't exist